core/src/mender-client.c

```diff
@@ -244,13 +251,17 @@ mender_err_t
 mender_client_init(mender_client_config_t *config, mender_client_callbacks_t *callbacks) {
 
     assert(NULL != config);
-    assert(NULL != config->device_type);
     assert(NULL != callbacks);
     assert(NULL != callbacks->restart);
     mender_err_t ret;
 
-    mender_client_config.device_type = config->device_type;
-
+    if ((NULL != config->device_type) && (0 != strlen(config->device_type))) {
```

Comment:
Let's add a macro `EMPTY_STRING()` checking for `!= NULL` and then checking `[0] == '\0'`. It's quite a bit cheaper then calling `strlen()`.